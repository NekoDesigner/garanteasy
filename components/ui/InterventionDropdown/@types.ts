/**
 * IGDropdown2Props - Generated by Garanteasier CLI
 */

export interface InterventionDropdownProps {
  style?: object;
  testID?: string;
}