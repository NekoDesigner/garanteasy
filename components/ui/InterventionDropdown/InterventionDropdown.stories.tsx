/**
 * GDropdown2 - Generated by Garanteasier CLI
 */

import React from "react";
import { View } from "react-native";
import GDropdown2 from "./index";

export default {
  title: 'GDropdown2',
  component: GDropdown2
};

export const Default = () => (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', padding: 16, gap: 24 }}>
      <GDropdown2 />
    </View>
);