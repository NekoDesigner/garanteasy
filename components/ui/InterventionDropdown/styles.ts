/**
 * GDropdown2Styles - Generated by Garanteasier CLI
 */

import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({});

export default styles;
