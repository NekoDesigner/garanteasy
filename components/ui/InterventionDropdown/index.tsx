import { AntDesign } from "@expo/vector-icons";
import React, { useState } from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import { History, HistoryDisplayLabel } from "../../../models/History/History";
import { InterventionDropdownProps } from "./@types";

export const INTERVENTION_OPTIONS = ["Réparation", "Maintenance", "Mise à jour", "Remplacement", "Inspection"] as const;

const InterventionDropdown: React.FC<InterventionDropdownProps> = ({ onValueChange }) => {
  const [open, setOpen] = useState(false);
  const [selected, setSelected] = useState("Réparation");

  const options = INTERVENTION_OPTIONS;

  const toggleDropdown = () => setOpen(!open);

  const handleSelect = (value: HistoryDisplayLabel) => {
    setSelected(value);
    setOpen(false);
    if (onValueChange) {
      onValueChange(History.setLabelToDatabaseFormat(value));
    }
  };

  return (
    <View style={styles.container}>
      {/* Bouton principal */}
      <TouchableOpacity style={styles.button} onPress={toggleDropdown} activeOpacity={0.7}>
        <Text style={styles.buttonText}>{selected}</Text>
        <AntDesign
          name={open ? "up" : "down"}
          size={16}
          color="#140E57"
          style={{ marginLeft: 6 }}
        />
      </TouchableOpacity>

      {/* Menu déroulant */}
      {open && (
        <View style={styles.dropdown}>
          {options.map((item) => (
            <TouchableOpacity key={item} style={styles.option} onPress={() => handleSelect(item)}>
              <Text style={styles.optionText}>{item}</Text>
            </TouchableOpacity>
          ))}
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: "100%",
    position: "relative",
  },
  button: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    borderWidth: 1,
    borderColor: "#140E57",
    borderRadius: 6,
    paddingVertical: 6,
    paddingHorizontal: 14,
    backgroundColor: "#fff",
    width: "100%",
  },
  buttonText: {
    fontSize: 14,
    color: "#140E57",
    fontWeight: "600",
  },
  dropdown: {
    position: "absolute",
    top: 40, // juste sous le bouton
    left: 0,
    right: 0,
    borderWidth: 1,
    borderColor: "#140E57",
    borderRadius: 6,
    backgroundColor: "#fff",
    zIndex: 1000,
  },
  option: {
    paddingVertical: 8,
    paddingHorizontal: 12,
  },
  optionText: {
    fontSize: 14,
    color: "#140E57",
  },
});

export default InterventionDropdown;
