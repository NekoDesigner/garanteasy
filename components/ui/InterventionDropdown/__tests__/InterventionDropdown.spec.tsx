/**
 * GDropdown2 - Generated by Garanteasier CLI
 */

import { render } from "@testing-library/react-native";
import React from "react";
import InterventionDropdown from "../index";

describe("GDropdown2 Components", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

    test('renders correctly', () => {
      const tree = render(<InterventionDropdown />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});