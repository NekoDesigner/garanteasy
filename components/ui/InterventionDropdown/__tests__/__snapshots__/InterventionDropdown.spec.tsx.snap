// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GDropdown2 Components renders correctly 1`] = `
<View
  style={
    {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#140E57",
        "borderRadius": 6,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 14,
        "paddingVertical": 6,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#140E57",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
      RÃ©paration
    </Text>
    <Text />
  </View>
</View>
`;
